pkg/recipes/driver/bicep.go

```diff
@@ -162,6 +163,12 @@ func (d *bicepDriver) Execute(ctx context.Context, opts ExecuteOptions) (*recipe
 
 	resp, err := poller.PollUntilDone(ctx, &runtime.PollUntilDoneOptions{Frequency: pollFrequency})
 	if err != nil {
+		errorDetails := recipes.GetErrorDetails(err)
+		for _, error := range errorDetails.Details {
+			if error.Code == "AuthorizationFailed" && providerConfig.Az == nil {
```

Comment:
I think there a few reasons why the DE might return `AuthorizationFailed`. And the error message here isn't right for all of them. 


Would it make sense to fix this in the DE? The DE already evaluates the template, so it would be easy to check this before even trying to create resources.